### PR TITLE
Extend ShowBgpAddressFamily class to parse additional BGP commands

### PR DESCRIPTION
**Pull request by Kani999**
## Description
Currently, the ShowBgpAddressFamily class in [this file](https://github.com/CiscoTestAutomation/genieparser/blob/882c298551531705697c097fa054f6573fa2c2de/src/genie/libs/parser/iosxr/show_bgp.py#L9091C1-L9091C27) parses commands show bgp and show bgp {address_family}. However, it needs to be extended to parse additional commands:

show bgp {address_family} community {community}
show bgp {address_family} community {community} {exact_match}
The objective is to ensure that the output of these new commands is compatible with the existing logic for parsing show bgp.

## Motivation and Context
Extend the ShowBgpAddressFamily class to parse the additional commands mentioned above.
Ensure that the output format of the new commands aligns with the output format of show bgp for compatibility with existing parsing logic.

## Related:
- Related Issue: #821 

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

**Comment by Kani999** [comment]
- Your regex pattern did not work correctly for IPv6 addresses. The regex in the address ignored the format [a-f]. 
---
Additionally, your code does not handle IPv6 outputs from BGP correctly.
Example:
```
s>i2001:718::2:99/128 195.113.156.4            0    100      0 i
s i                   195.113.156.4            0    100      0 i
s>i2001:718::2:101/128
                      10.2.8.1                 0    100      0 i
s i                   10.2.8.1                 0    100      0 i
s>i2001:718::2:116/128
                      195.113.156.9         1000    205      0 65086 i
s i                   195.113.156.9         1000    205      0 65086 i
s i                   2001:718:0:600:0:1a:1a:11
                                            1000    110      0 65086 i
```
- [Your current regex](https://regex101.com/r/w6F0eT/1)
- [My Edit](https://regex101.com/r/18lHd3/1)

In this state, it completely ignores that the next hop and metrics are listed on the next line. My regex corrects this, **however, there is a flaw in your code**: you use a multiline regex but execute it on individual lines, which does not make sense. To make the code work correctly now, I need to modify the device output to this format:

```
s>i2001:718::2:99/128 195.113.156.4            0    100      0 i
s i                   195.113.156.4            0    100      0 i
s>i2001:718::2:101/128                      10.2.8.1                 0    100      0 i
s i                   10.2.8.1                 0    100      0 i
s>i2001:718::2:116/128                      195.113.156.9         1000    205      0 65086 i
s i                   195.113.156.9         1000    205      0 65086 i
s i                   2001:718:0:600:0:1a:1a:11                                            1000    110      0 65086 i
```
- [Your current regex](https://regex101.com/r/OUMOf9/2)
- [My Edit](https://regex101.com/r/vTCRGp/1)

This is achieved with the following code:
```python
    raw_result = device.execute(command)        
    raw_result = raw_result.replace('\r\n', '\n')    
    lines = raw_result.split('\n')
    result = [lines[0]]

    for line in lines[1:]:
        if line.startswith(' '):
            result[-1] += line
        else:
            result.append(line)

    formatted_text = '\n'.join(result)

   # Run GENIE with preprocessed formatted text
   device.parse(command, output=formatted_text)
```

This code concatenates lines that start with a space to the previous line.

With this output modification and my changes in the pull request, the "show bgp" command works for both IPv4 and IPv6 addresses. Currently, your implementation for IPv6 is not functional.

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_bgp.py:9172]: Please comment above this what you're now matching

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_bgp.py:8670]: Please comment above this what you're now matching

**Review by ThomasJRyan — CHANGES_REQUESTED** [review]
Hey there! Sorry for the delay in reviewing. This looks good. However, we'll need a changelog file and unittest included before we can merge it in. You can view other merged PRs for examples of them! Be sure to request a re-review once you've added them in

**Review by ThomasJRyan — CHANGES_REQUESTED** [review]
Hello,

I'm sorry, but I cannot approve this without an accompanying unittest. Please have a look at other PRs to see how to add one

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (domachad) [reply src/genie/libs/parser/iosxr/show_bgp.py:8699]: add an example that requires this change on top of the pattern declaration

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (domachad) [reply src/genie/libs/parser/iosxr/show_bgp.py:9201]: same here

**Review by domachad — CHANGES_REQUESTED** [review]
please add at least one unit test to support your changes

**Comment by Kani999** [comment]
> please add at least one unit test to support your changes

Tests were added

**Review by Kani999 — COMMENTED** [review]
Unit test were added in the last commit

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (SohanTirpude) [reply src/genie/libs/parser/iosxr/show_bgp.py:9203]: @Kani999, kindly do not resolve the conversion unless you address the comments. You have to add the example for which you have made the changes in regex pattern. Kindly do so and then only resolve the conversion.

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (SohanTirpude) [reply src/genie/libs/parser/iosxr/show_bgp.py:8699]: @Kani999, kindly do not resolve the conversion unless you address the comments. You have to add the example for which you have made the changes in regex pattern. Kindly do so and then only resolve the conversion.

Also, if you have made changes to `ShowBgpVrf` parser class then you must add the unit tests like you added for `ShowBgpAddressFamily`. Kindly do so and then resolve the conversation.

Comment (Kani999) [reply src/genie/libs/parser/iosxr/show_bgp.py:8706]: Ok, Difference between Old and New regex:

- Old one: https://regex101.com/r/w6F0eT/1
- New One: https://regex101.com/r/18lHd3/1

File: src/genie/libs/parser/iosxr/show_bgp.py
Comment (Kani999) [reply src/genie/libs/parser/iosxr/show_bgp.py:9210]: Ok, Difference between Old and New regex:
-    Old one: https://regex101.com/r/w6F0eT/1
-    New One: https://regex101.com/r/18lHd3/1

**Review by ThomasJRyan — APPROVED** [review]

**Pull request merged**
